### PR TITLE
[WIP] Cake support

### PR DESCRIPTION
**Pull request by mholo65**
Adding support for Cake (I think). Honestly, I have no idea what I'm doing 😄 @gep13 This needs some TypeScript love from you.

OmniSharp-Roslyn PR here https://github.com/OmniSharp/omnisharp-roslyn/pull/932

NB! For some reason, you'll have to switch language mode from `Cake` to `C#` (If you have Cake VS Code Extension installed) to get intellisense.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -460,13 +462,11 @@
             "razor"
           ]
         },
-        "runtime": "node",
```
Comment (DustinCampbell) [inline package.json]: You don't want this change

File: package.json [hunk 2]
```diff
@@ -460,13 +462,11 @@
...
         "runtimeArgs": [],
         "variables": {
           "pickProcess": "csharp.listProcess",
           "pickRemoteProcess": "csharp.listRemoteProcess"
         },
-        "program": "./out/src/coreclr-debug/proxy.js",
```
Comment (DustinCampbell) [inline package.json]: Or this one

File: package.json [hunk 3]
```diff
@@ -1376,7 +1376,16 @@
...
+        "windows": {
+          "program": "./.debugger/vsdbg-ui.exe"
+        },
+        "osx": {
+          "program": "./.debugger/vsdbg-ui"
+        },
+        "linux": {
+          "program": "./.debugger/vsdbg-ui"
+        }
       },
```
Comment (DustinCampbell) [inline package.json]: Or these

File: package.json [hunk 4]
```diff
@@ -1387,13 +1396,11 @@
             "razor"
           ]
         },
-        "runtime": "node",
```
Comment (DustinCampbell) [inline package.json]: Not this

File: package.json [hunk 5]
```diff
@@ -1387,13 +1396,11 @@
...
         "runtimeArgs": [],
         "variables": {
           "pickProcess": "csharp.listProcess",
           "pickRemoteProcess": "csharp.listRemoteProcess"
         },
-        "program": "./out/src/coreclr-debug/proxy.js",
```
Comment (DustinCampbell) [inline package.json]: nor this

File: package.json [hunk 6]
```diff
@@ -2156,6 +2163,15 @@
...
           }
+        },
+        "windows": {
+          "program": "./.debugger/vsdbg-ui.exe"
+        },
+        "osx": {
+          "program": "./.debugger/vsdbg-ui"
+        },
+        "linux": {
+          "program": "./.debugger/vsdbg-ui"
```
Comment (DustinCampbell) [inline package.json]: Or this

File: package.json (hunk 1 above)
Comment (mholo65) [inline package.json]: Oh, are VSCode adding these changes automatically? Can't remember changing these.

Comment (DustinCampbell) [inline package.json]: Yes, this happens when you debug the extension.

**Comment by gep13** [comment]
@mholo65 was there something in particular that you thought was "wrong" with this PR?  The changes look good to me. :+1:

**Comment by mholo65** [comment]
@gep13 well if we could figure out why we have to change language mode from `Cake` to `C#` before this extension wants to send any requests to `omnisharp.exe`, that would be great :wink:

`omnisharp.exe` is started when a`.cake` file is opened, but no requests are sent. Current fix is to switch language mode

**Comment by DustinCampbell** [comment]
> well if we could figure out why we have to change language mode from `Cake` to `C#` before this extension wants to send any requests to `omnisharp.exe`, that would be great :wink:

I posted how that happens in the #cake channel on Slack. The file types that trigger "csharp" as the language is defined in VS Code itself right [here](https://github.com/Microsoft/vscode/blob/master/extensions/csharp/package.json#L16-L19). I'll submit a change to add "*.cake" to this list on your behalf today.

**Comment by gep13** [comment]
>@DustinCampbell said...
The file types that trigger "csharp" as the language is defined in VS Code itself right here. I'll submit a change to add "*.cake" to this list on your behalf today.

Yes, I saw that in the Slack channel, and that is why I raised this issue https://github.com/OmniSharp/omnisharp-vscode/issues/1684, I was trying to get clarity on whether the syntax highlighting that we provide for .cake files would be affected by making that change.

Do you know if this will be the case?

**Comment by DustinCampbell** [comment]
Yes, the syntax highlighting for .cake files would be affected. What differences does Cake add to C# syntax highlighting? Note that the official C# support for cake highlighting is here: https://github.com/dotnet/csharp-tmLanguage.

**Comment by DustinCampbell** [comment]
I think this is generally a problem of trying to spread Cake support across two extensions. If you're trying to have a Cake Support extension provide syntax highlighting *and* the C# extension, I'm less comfortable with taking this change.

**Comment by DustinCampbell** [comment]
Cake is really just C# script with some semantic differences, correct? If that's the case, I would proposal that the Cake Support extension doesn't provide it's own syntax highlighting, but takes advantage of C#'s. Then, the Cake Support extension would be modified to operate on the "csharp" language but only when the extension is ".cake". Thoughts?

**Comment by DustinCampbell** [comment]
I'm going to go ahead and move this conversation over to #1684.

**Comment by DustinCampbell** [comment]
This looks good. Sorry for the delay!

**Comment by gep13** [comment]
WOO HOO! Thanks for merging this!

**Pull request merged**